core/src/main/scala/no/nrk/bigquery/BQPartitionType.scala

```diff
@@ -21,24 +21,36 @@ import java.time.{LocalDate, YearMonth}
   *
   * It is meant that when we add support for more partition schemes, the first step is to add a new subtype here.
   */
-sealed trait BQPartitionType[+Param]
+sealed trait BQPartitionType[+Param] {
+  def fieldOpt: Option[Ident]
```

Comment:
since we are in a sealed trait we can do this:
```
    def partitionField: Option[Ident] = typ match {
      case BQPartitionType.DatePartitioned(field)         => Some(field)
      case BQPartitionType.MonthPartitioned(field)      => Some(field)
      case BQPartitionType.RangePartitioned(field, _)   => Some(field)
      case sharded: BQPartitionType.Sharded                 => None
      case partitioned: BQPartitionType.NotPartitioned => None
    }
```